### ISSUE TITLE
Provenance of mapping review

### ISSUE THREAD

**Issue by mellybelly** [mapping-commons/sssom#106]
From @kec

Regarding the mapping has fields for author_id, reviewer_id, and creator_id. It may be that there is a need for more than one workflow, particularly for things that might have patient safety or other highly reliable organization implications, for example: 

Authoring with no review
Authoring with sequential single review
Independent authoring with 3rd party adjudication
Double blind review
and more…

Having these author_id, reviewer_id, and creator_id fields hard-coded in the data structure limits the ability to represent these other types of review.  It maybe useful to have a review field with a small CV available for the review status.

**Comment by mellybelly** [comment]
An alternative might be to split a "reference" mapping (file) and then have "in progress instances" where you could associate author, reviewer as metadata, etc. Instances would need a path to promote to “reference". This would assume the reference is published, and not still in the pipeline.

**Comment by kec** [comment]
One way of thinking about the split could be a “snapshot” file that has a summary of the chosen information at a particular point in time, as well as a “full” file, that contains the history of the object from which all snapshots can be derived.

**Comment by matentzn** [comment]
@mellybelly yes, this is stretching the applicability of SSSOM - our leading design principle is simplicity, which in particular (despite our hearts yearning for it) avoids nested structures that would enable combining multiple types of reviews in a single mapping set. We could easily enable more powerful provenance by simply adding workflow "nodes" into the model (authoring, review etc) that all get their own metadata - but then we leave the realm of the flat table structure behind, which is one of the 4 design requirements for SSSOM. 

I would agree to solve this at the level of differently scoped mapping sets. However, as an alternative, you can use the `other` slot to convey such information: you can basically carry arbitrary metadata in there. While this may not be ideal, its a design principle I adopted from @ShahimEssaid - later, when enough people use the same "system", we can consider a "complex" extension to sssom that allows for nested structures and, therefore, `review_type` as well. 

@kec thank you for your input - would you perhaps have the time to describe your scenario in a bit more detail? Maybe we can find a more elegant way to address it.

**Comment by matentzn** [comment]
@mellybelly @kec - was there still interest on trying to make progress on this?

**Comment by kec** [comment]
Moving to my FDA address, as my other address is overwhelmed by spam.

With my activities at FDA, we are working on implementing knowledge management capabilities that include integration of overlapping systems, and support for sssom can be one of those capabilities. I’d like to encourage that we try and align under the Tinkar efforts at HL7:

Guidelines for a Standardized Terminology Knowledge base Project - Vocabulary - Confluence (hl7.org)<https://confluence.hl7.org/display/VOC/Guidelines+for+a+Standardized+Terminology+Knowledge+base+Project>

I have interest in advancing progress within this project or in collaboration with this project.


Begin forwarded message:

**Comment by matentzn** [comment]
Ok, just for the record, while terminologically slightly counterintuitive, I think the best way to capture reviews is to record them as separate mapping_justifications. So rather than reviewing a justification, we review a mapping (which is in any case closer to how most people would think of a "mapping review"). So I believe we can already capture the review process reasonably well. Could you perhaps provide a list of "review status" use cases, so we can start thinking about how to represent these perfectly? A list with bullet points with examples would be best!
